hbase_storage/hbase1.1.2.2.5/src/main/java/com/splicemachine/storage/ClientPartition.java

```diff
@@ -153,32 +153,26 @@ public boolean checkAndPut(byte[] key,byte[] family,byte[] qualifier,byte[] expe
             List<Partition> partitions;
             if (!refresh) {
                 partitions = partitionInfoCache.getIfPresent(tableName);
-                if (partitions == null) {
-                    partitions = formatPartitions(getAllRegionLocations(false));
-                    assert partitions!=null:"partitions are null";
-                    partitionInfoCache.put(tableName, partitions);
+                if (partitions != null) {
+                    return partitions;
                 }
-                return partitions;
             }
-            partitions = formatPartitions(getAllRegionLocations(true));
-            partitionInfoCache.invalidate(tableName);
-            assert partitions!=null:"partitions are null";
-            partitionInfoCache.put(tableName,partitions);
+            List<HRegionLocation> tableLocations = ((ClusterConnection) connection).locateRegions(tableName, !refresh, false);
+            partitions = new ArrayList<>(tableLocations.size());
+            for (HRegionLocation location : tableLocations){
+                if (location.getServerName() == null) {     // ensure no offline regions
+                    continue;
+                }
```

Comment:
Does skipping offline regions ever result in a case where we miss data? That is, is tableLocations ever used in a way that ignores a region that's temporarily down (due to an RS failure, say) such that the missing region doesn't get detected and reported later?